graylog2-web-interface/src/components/graylog/Panel.jsx

```diff
@@ -7,59 +9,232 @@ import teinte from 'theme/teinte';
+  border-top-color: ${teinte.secondary.due};
+`;
 
-      + .panel-collapse > .panel-body {
-        border-top-color: ${borderColor};
-      }
-      .badge {
-        color: ${backgroundColor};
-        background-color: ${hex};
-      }
+const panelVariantStyles = hex => css`
+  border-color: ${borderColor(hex)};
+
+  & > ${PanelHeading} {
+    color: ${colorLevel(backgroundColor(hex), 9)};
+    background-color: ${backgroundColor(hex)};
+    border-color: ${borderColor(hex)};
+
+    > .panel-title,
+    > .panel-title > * {
+      font-size: 16px;
     }
-    & > .panel-footer {
-      + .panel-collapse > .panel-body {
-        border-bottom-color: ${borderColor};
-      }
+
+    + .panel-collapse > .panel-body {
+      border-top-color: ${borderColor(hex)};
     }
-  `;
-};
+
+    .badge {
+      color: ${backgroundColor(hex)};
+      background-color: ${hex};
+    }
+  }
+
+  & > ${PanelFooter} {
+    + .panel-collapse > .panel-body {
+      border-bottom-color: ${borderColor(hex)};
+    }
+  }
+`;
 
 const StyledPanel = styled(BootstrapPanel)`
   background-color: ${teinte.primary.due};
 
+  .panel-group {
+    ${PanelHeading} {
+      + .panel-collapse > .panel-body,
+      + .panel-collapse > .list-group {
+        border-top-color: ${teinte.secondary.due};
+      }
+    }
+
+    ${PanelFooter} {
+      + .panel-collapse .panel-body {
+        border-bottom-color: ${teinte.secondary.due};
+      }
+    }
+  }
+
+  ${bsStyleThemeVariant(panelVariantStyles)};
+`;
+
+const deprecatedVariantStyles = hex => css`
+  /** NOTE: Deprecated & should be removed in 4.0 */
+  border-color: ${borderColor(hex)};
+
+  & > .panel-heading {
+    color: ${colorLevel(backgroundColor(hex), 9)};
+    background-color: ${backgroundColor(hex)};
+    border-color: ${borderColor(hex)};
+
+    > .panel-title,
+    > .panel-title > * {
+      font-size: 16px;
+    }
+
+    + .panel-collapse > .panel-body {
+      border-top-color: ${borderColor(hex)};
+    }
+    .badge {
+      color: ${backgroundColor(hex)};
+      background-color: ${hex};
+    }
+  }
+
+  & > .panel-footer {
+    + .panel-collapse > .panel-body {
+      border-bottom-color: ${borderColor(hex)};
+    }
+  }
+`;
+
+const DeprecatedStyledPanel = styled(BootstrapPanel)`
+  /** NOTE: Deprecated & should be removed in 4.0 */
+  background-color: ${teinte.primary.due};
 
   .panel-footer {
     background-color: ${teinte.secondary.tre};
     border-top-color: ${teinte.secondary.due};
   }
-
   .panel-group {
     .panel-heading {
       + .panel-collapse > .panel-body,
       + .panel-collapse > .list-group {
         border-top-color: ${teinte.secondary.due};
       }
     }
-
     .panel-footer {
       + .panel-collapse .panel-body {
         border-bottom-color: ${teinte.secondary.due};
       }
     }
   }
 
-  ${bsStyleThemeVariant(panelVariantStyles)}
+  ${bsStyleThemeVariant(deprecatedVariantStyles)}
 `;
 
-export default StyledPanel;
+const Panel = ({
```

Comment:
Some old `Panel` components still look pretty bad after the changes:

<img width="1009" alt="Screenshot 2019-12-17 at 17 54 39" src="https://user-images.githubusercontent.com/716185/71017175-7e3a1200-20f6-11ea-939d-7fbd77683218.png">
<img width="974" alt="Screenshot 2019-12-17 at 17 54 44" src="https://user-images.githubusercontent.com/716185/71017182-809c6c00-20f6-11ea-9dd0-8a7eba8869fc.png">

I guess at this point is better to adapt the legacy component so they look better, but if it's too hard we can also look through the usages and fix them.

Reply:
Next commit will have these (and others) worked out

![image](https://user-images.githubusercontent.com/122591/71199304-45677d80-225b-11ea-99ee-a83ee0b82783.png)